docs: streamline A2UI v0.9.1 architecture skill, add AGENTS.md and maintenance skill (#1542)
google/A2UI: .gemini/skills/a2ui-agent-maintenance/SKILL.md

### 3. Authority of the Specifications

When updating agent guides or skills, you must use the latest active specifications under `specification/` as your absolute source of truth. Do not make assumptions. Read these files in full using available context retrieval tools:

- **Protocol & Message Semantics:** `specification/v0_9_1/docs/a2ui_protocol.md`
- **Renderer & Subscription Lifecycles:** `specification/v0_9_1/docs/renderer_guide.md`
- **JSON Schemas:** `specification/v0_9_1/json/*.json`
- **Layouts, Components, & Functions:** `specification/v0_9_1/catalogs/basic/catalog.json`

### 4. Use Subagents to Explore and Summarize

When updating documentation or skills for other parts of the codebase (such as client libraries, agent SDKs, or demos), use specialized subagents or helper assistants to inspect and summarize those regions. For example:

- _Client Libraries:_ Invoke a helper agent to check for new packages, configurations (`package.json`), or initialization flows.
- _Sample Apps:_ Invoke a helper agent to test or check if running scripts (e.g. `uv run`, `npm run dev`) have changed.

---

> ## Mandatory Maintenance Workflow

Reviewer: Is the workflow mandatory, or the maintenance?

Author: I think we can remove "mandatory" from this line.

When a task requires updating or synchronizing the agent instructions:

### Step 1: Research the Repo (Subagent Delegation)

Delegate the codebase inspection to a helper or subagent to get an objective summary of recent changes. Instruct the helper/subagent with a clear prompt:

> _"Inspect the repository structure, specifically identifying any changes to directories, client libraries, samples, or tools. Compare the actual files found against the current listings in `.gemini/AGENTS.md` to identify discrepancies."_

### Step 2: Formulate the Strategy

Review the findings compiled by your helper agent. Identify:

- New features or directories to document.

> - Deprecated files or steps to remove.

Reviewer: Maybe other significant changes that aren't additions or removals?

Author: Agreed - especially changes to the specification, major SDK features, changes to the SDK

> - Setup or build scripts that need updating in the running instructions.

Reviewer: Maybe add documentation that has been updated?

Author: Maybe we can mention "changes to the specification" specifically? I worry about saying documentation overall, because a lot of our docs are stale, and I think this content should be updated directly from the specification / code sources of truth.

### Step 3: Write to the Source of Truth (`AGENTS.md`)

> Make surgical updates to `.gemini/AGENTS.md`. Never use vague or flowery language (e.g., "significantly enhances", "powerful tool"). Keep the style factual, technical, generic, and concise.

Reviewer: I have an entire skill for "natural writing" that tries to counteract these kinds of things.  Maybe we should just add that as a separate skill here?

Author: SGTM, let's remove the stuff about writing style here and add your skill!

Let's keep this heading though, and mention updating both AGENTs.md and any relevant skills.

Let's also note here that the agent documentation should primarily point agents to the most important sources of truth (e.g. docs, code etc) and skills should provide step-by-step recipes for completing certain tasks, but it should be brief and avoid duplicating information from the sources of truth e.g. specification and SDKs. I think this guidance here would be specific to agent documentation and might not be included in "natural writing".

Reviewer: Sure.  That's separate from not using flowery sentences or bullet point lists.

### Step 4: Keep Entry-Point Gatekeepers Simple

Any entry-point gatekeeper file (such as `.gemini/GEMINI.md` or equivalent) must remain a simple direct-import or folder-pointer that directs agents to read `AGENTS.md`. It must not contain or duplicate details already maintained inside `AGENTS.md`.

### Step 5: Post-Task Suggestion Rule

> Never automatically commit updates to `AGENTS.md` or skills unless explicitly requested by the user. Always present your recommended changes as a clean suggestion at the end of your response, explaining _why_ the update is needed.

Reviewer: This is a tiny nit:

```suggestion
Never automatically commit updates to `AGENTS.md` or skills unless explicitly requested by the user. Always present your recommended changes as a suggestion in your response, explaining _why_ the update is needed.
```

I worry that it will compete with other things in the response if there are multiple suggestions and they're all vying for the end position.

Author: Yeah, that makes sense - I like your proposal!
